Mage.Sets/src/mage/cards/h/HashatonScarabsFist.java

```diff
@@ -0,0 +1,92 @@
+package mage.cards.h;
+
+import java.util.Optional;
+import java.util.UUID;
+
+import mage.MageInt;
+import mage.ObjectColor;
+import mage.abilities.Ability;
+import mage.abilities.costs.mana.ManaCostsImpl;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.CreateTokenCopyTargetEffect;
+import mage.abilities.effects.common.DiscardCardControllerTriggeredAbility;
+import mage.abilities.effects.common.DoIfCostPaid;
+import mage.constants.Outcome;
+import mage.constants.SubType;
+import mage.constants.SuperType;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.filter.StaticFilters;
+import mage.game.Game;
+import mage.players.Player;
+import mage.target.targetpointer.FixedTarget;
+
+/**
+ * @author sobiech
+ */
+public final class HashatonScarabsFist extends CardImpl {
+
+    public HashatonScarabsFist(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{W}{B}");
+
+        this.supertype.add(SuperType.LEGENDARY);
+        this.subtype.add(SubType.ZOMBIE);
+        this.subtype.add(SubType.WIZARD);
+        this.power = new MageInt(1);
+        this.toughness = new MageInt(3);
+
+        // Whenever you discard a creature card, you may pay {2}{U}. If you do, create a tapped token that's a copy of that card, except it's a 4/4 black Zombie.
+        final Ability ability = new DiscardCardControllerTriggeredAbility(
+                new DoIfCostPaid(new HashatonScarabsFistEffect(), new ManaCostsImpl<>("{2}{U}")), false, StaticFilters.FILTER_CARD_CREATURE_A
+        );
+        this.addAbility(ability);
+    }
+
+    private HashatonScarabsFist(final HashatonScarabsFist card) {
+        super(card);
+    }
+
+    @Override
+    public HashatonScarabsFist copy() {
+        return new HashatonScarabsFist(this);
+    }
+}
+
+class HashatonScarabsFistEffect extends OneShotEffect {
+
+    HashatonScarabsFistEffect() {
+        super(Outcome.PutCreatureInPlay);
+        this.staticText = "create a tapped token that`s a copy of that card, except it`s a 4/4 black Zombie";
```

Comment:
you can try to override effect’s gettext  and add info from this.values.get("discardedCard") if exists (after super.gettext). Maybe it will work for choose dialog (but not sure about stack/logs info).

Reply:
![IMG_1118](https://github.com/user-attachments/assets/f7cdcdf1-4ef5-483c-a273-c65f19d04555)

Reply:
Alternative - add custom card hint StaticHint and return that info on exists (this.values.get("discardedCard")). It’s a better solution.